src/averageAges.js

```diff
@@ -55,7 +71,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : people;
+
+  const diffs = children.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    if (mother) {
+      return child.born - mother.born;
+    } else {
+      return undefined;
+    }
+  }).filter(diff => diff !== undefined);
+
+  const sumOfDiffs = diffs.reduce((total, diff) => {
+    return total + diff;
+  }, 0);
```

Comment:
let's use less array methods))

Reply:
I got it) i'll refactor code to using less duplicates